### PR TITLE
Add SKIPPED test output status for SAST tasks

### PR DESCRIPTION
**Pull request by jinqi7**
Signed-off-by: Jing Qi <jinqi@redhat.com>

It's to add SKIPPED status for the SAST tasks

### CODE REVIEW

File: tasks/sast-go.yaml [hunk 1]
```diff
@@ -21,31 +21,37 @@ spec:
...
       script: |
-        /usr/local/go/bin/gosec -no-fail -fmt=sarif -out=gosec_output.json $(workspaces.workspace.path)/... | tee gosec_output.txt
+        /usr/local/go/bin/gosec -no-fail -fmt=sarif -out=gosec_output.json $(workspaces.workspace.path)/... 2> gosec_output.txt
 
         # Test if any package was found
         # Even with -no-fail, gosec uses exit code 1 for several states,
         # including when there are no packages found.
-        grep "No packages found$" gosec_output.txt && echo "Skipping because no Go packages were found"
+        SKIP_MSG="No packages found"
+        grep -q "$SKIP_MSG" gosec_output.txt
```
Comment (Michkov) [inline tasks/sast-go.yaml:30]: you are not changing shebang so it's using default `#!/bin/bash -xe`, so when this grep fails then it stops execution.

Comment (jinqi7) [reply tasks/sast-go.yaml]: Yes. Updated the scripts.

**Comment by jinqi7** [comment]
@Michkov Anything changed in the  e2e-tests? It can't pass with even a small change in the script.

**Review by Dannyb48 — APPROVED** [review]
LGTM

File: tasks/sast-java-sec-check.yaml [hunk 2]
```diff
@@ -50,23 +50,30 @@ spec:
...
+          SKIP_MSG="jar file $JAR_PATH doesn't exist"
+          test_skipped=1
         fi
 
         ERR_MSG="$(cat stderr.txt)"
         ERR_MSG="${ERR_MSG:-unknown}"
-        HACBS_ERROR_OUTPUT=$(jq -rc --arg date $(date +%s) --arg ERR_MSG "${ERR_MSG: 0: 3000}" --null-input \
-          '{result: "ERROR", timestamp: $date, failures: [$ERR_MSG]}')
-        HACBS_TEST_OUTPUT=$(jq -rce --arg date $(date +%s) \
+        HACBS_ERROR_OUTPUT=$(jq -rc --arg date $(date +%s) --arg MSG "${ERR_MSG}" --arg test_skipped $test_skipped \
```
[the remark this answers is not in the document]
Comment (jinqi7) [reply tasks/sast-java-sec-check.yaml]: OK. Added it back.

File: tasks/sast-java-sec-check.yaml [hunk 3]
```diff
@@ -50,23 +50,30 @@ spec:
...
 
-        JAR_PATH=$(workspaces.workspace.path)/$(params.PATH_CONTEXT)/target/*.jar
+        JAR_PATH=`ls $(workspaces.workspace.path)/$(params.PATH_CONTEXT)/target/*.jar`
         if [ -f "$JAR_PATH" ]; then
           /home/findsecbugs-cli/findsecbugs.sh $(params.OPTIONAL_ARGS) $(params.OUTPUT_ONLY_ANALYZE) -$(params.OUTPUT_FORMAT) \
             -output sast_java_sec_output.json $JAR_PATH  2> stderr.txt
+          test_skipped=0
         else
-          echo "jar file $JAR_PATH doesn't exist" > stderr.txt
+          SKIP_MSG="jar file $JAR_PATH doesn't exist"
```
[the remark this answers is not in the document]
Comment (jinqi7) [reply tasks/sast-java-sec-check.yaml]: Updated

**Comment by jinqi7** [comment]
@staticf0x , could you please help to review it ?

File: tasks/sast-go.yaml (hunk 1 above)
Comment (jinqi7) [reply tasks/sast-go.yaml]: @Michkov , could please help to review it again?

Comment (jinqi7) [reply tasks/sast-go.yaml]: @Michkov , could you please help to see the e2e-tests issue?

**Comment by Michkov** [comment]
@psturc Could you check why the e2e failing?

**Comment by jinqi7** [comment]
@psturc Could you please help to see the e2e-tests failure of " open /root/tests/e2e-demos/config/default.yaml: no such file or directory" here? Thanks

**Comment by psturc** [comment]
> @psturc Could you please help to see the e2e-tests failure of " open /root/tests/e2e-demos/config/default.yaml: no such file or directory" here? Thanks

👀

**Comment by psturc** [comment]
I submitted a PR with a fix (see above). Will let you know once your PR is ready to retest

**Comment by psturc** [comment]
@jinqi7  PR merged, new image pushed, feel free to run the tests again

**Review by hongweiliu17 — APPROVED** [review]
LGTM

**Pull request merged**